internal/transport/http2_server.go

```diff
@@ -1062,11 +1066,7 @@ func (t *http2Server) WriteStatus(s *Stream, st *status.Status) error {
 func (t *http2Server) Write(s *Stream, hdr []byte, data []byte, opts *Options) error {
 	if !s.isHeaderSent() { // Headers haven't been written yet.
 		if err := t.WriteHeader(s, nil); err != nil {
-			if _, ok := err.(ConnectionError); ok {
-				return err
-			}
-			// TODO(mmukhi, dfawley): Make sure this is the right code to return.
-			return status.Errorf(codes.Internal, "transport: %v", err)
+			return err
```

Comment:
Is it safe to do this or should this always wrap the error in `status.Errorf`? I'm not sure whether we always return a `*Status` error from inside `WriteHeader` or whether there's a convention we adhere to here. 

It looks like `writeHeaderLocked` can return a bare error here: https://github.com/grpc/grpc-go/blob/master/internal/transport/http2_server.go#L949. Should that be wrapped with an `INTERNAL` code or is it fine to return that as is if `ErrHeaderListSizeLimitViolation` is changed as well?

Reply:
> Is it safe to do this

We control the caller, so it should be fine.

> It looks like writeHeaderLocked can return a bare error here

Yes, let's make sure the `WriteHeader` function _always_ returns a status error.

Reply:
Cool - done.